src/index.js

```diff
@@ -1,97 +1,62 @@
-// END GOAL:
-// const selector = select`profile.name.last`;
-
-// function reducer(state, action) {
-//   const lastNameSelector = selector(state);
-//   // getting the value is handy in components and mapStateToProps
-//   let currentVal = lastNameSelector.get();
-
-//   // But I think its more handy for setting
-//   lastNameSelector.set(action.payload);
-
-//   // if it was an array value
-//   lastNameSelector.push("foo");
-//   let someIndex = action.payload;
-//   lastNameSelector.remove(someIndex);
-
-//   return lastNameSelector.save();
-// }
-
-export default function select(strings, ...exprs) {
-  const woven = strings.raw ? interleave(strings, ...exprs) : strings
-
-  // console.log(woven)
-
-  let key = woven.filter(Boolean)
-
-  // console.log('key', key)
-
-  let INITIAL = Symbol('INITIAL')
-  let oldObj
-  let prev = INITIAL
-
-  function selectorFn(obj) {
-    // console.log('obj', obj, 'key', key)
-    // from dlv https://raw.githubusercontent.com/developit/dlv/master/index.js
-    let p = 0
-
-    if (obj == null) {
-      return obj
+import 'setimmediate';
+
+// Used to transfer object from a `toPrimitive` call
+// to the recieving retarget
+let transfer;
+
+const removeTransferOnNextTick = () => {
```

Comment:
Why is this needed?

Reply:
If for some reason someone would concat a selector function with a string or use any operator on it it add itself as a transfer. Because I am abusing `String.toPrimitve` to have this `retarget.a[retarget.b]` equal `retarget.a.b`. This trick would remove the transfer after the next tick. 

To change this I have an idea that would slightly alter the syntax for retarget composition. It looks like this:

`retarget.a(retarget.b) => retarget.a.b`  

This solution would enable us to write a babel plugin that would transpile retarget proxies and make it work in any browser.